util/metrics/import_metrics_files.rb

```diff
@@ -0,0 +1,181 @@
+# frozen_string_literal: true
+
+require "optparse"
+
+# This script is intended to be run on a puppet infrastructure node that had the puppet_metrics_collector
+# module installed and running.  Given a time window it will collect the json files generated by the
+# puppet_metrics_collector module and tar them into the tar file.  This includes jsons that are in tar
+# files already.  Those will be extracted and included if they are in the time window.
+
+# General namespace for metrics module
+module Metrics
+  # Main class for collecting metrics files into a tar file
+  #
+  # @author Randell Pelak
+  #
+  # @attr [integer] start_epoch Epoch seconds for the start of the collection window
+  # @attr [integer] end_epoch Epoch seconds for the end of the collection window
+  # @attr [string] metrics_dir The "output directory" of the metrics collector puppet module
+  # @attr [string] output_dir The directory to place the tarfile in.
+  # @attr [string] tar_file_name Name of the tar file to create with the metrics
+  # @attr [boolean] verbose Verbose output
+  # @attr [integer] poll_interval The "collection_frequency" of the metrics collector puppet module
+  # @attr [integer] archive_interval The frequency of the *_metrics_dir cron job for the metrics
+  #   collector module.  This is how often it tars up the json files.
+  #
+  class ImportMetricsFiles
+    # Initialize class
+    #
+    # @author Bill Claytor
+    #
+    # @param [string] results_dir The GPLT results directory to process
+    # @param [string] prefix The prefix to use when building the server tag
+    # @param [string] json2graphite_path The path to the json2graphite.rb script
+    #
+    # @return [void]
+    #
+    # @example
+    #   initialize(results_dir, prefix)
+    #
+    def initialize(results_dir, prefix, json2graphite_path)
+      @results_dir = results_dir
+      @prefix = prefix
+      @json2graphite_path = json2graphite_path
+      @id = @results_dir.split("_").last
+    end
+
+    # The main entry point to the import_metrics_files script
+    #
+    # @author Bill Claytor
+    #
+    # @return [void]
+    #
+    # @example
+    #   import_metrics_files
+    #
+    def import_metrics_files
+      pmc_dir = "#{@results_dir}/puppet-metrics-collector"
+      raise "Directory not found: #{pmc_dir}" unless File.directory? pmc_dir
+
+      puts "Importing puppet_metrics_collector files:"
+      puts " results_dir: #{@results_dir}"
+      puts " prefix: #{@prefix}"
+      puts " id: #{@id}"
+
+      service_dirs = Dir.glob("#{pmc_dir}/*").select { |f| File.directory? f }
+
+      puts " service directories:"
+      puts service_dirs
+      puts
+
+      # each service directory
+      service_dirs.each do |service_dir|
+        puts "Checking directory: #{service_dir}"
```

Comment:
It is still going to output the info from the json2graphite script so you would really only see these as they're scrolling by or if you scroll up to review the output. And if you do that these server to break that up so you can see the order of execution. Is there a benefit to hiding this info and making the user specify an option to see it?

Reply:
It just about doing the right thing, even if the subscript isn't. :) but I can agree since the subscript is already spamming it probably doesn't matter.